contrib/dist/linux/buildrpm.sh

```diff
@@ -45,12 +88,69 @@ build_multiple=${build_multiple:-"no"}
 #########################################################################
 
 #
+# save original parameters
+#
+orig_param="$@"
+
+#
+# usage information
+#
+usage="Usage: $0 [-b][-o][-m][-d][-u][-h] [-n <name>][-f <lf_location>][-t <tm_location>] <tarball>
+
+	-b                do NOT build SRPM, only all-in-one binary RPM (required for other flags to work)
+	-n <name>         name of the resulting RPM package (openmpi)
+	-o                install in /opt/<name>/<version>
+	-m                create modulefiles
+	-d                build debuginfo RPM
+	-u                include usNIC support
+	-f <lf_location>  include LibFabric support from <lf_location>
+	-t <tm_location>  include Torque support from <tm_location>
+	-h                print this message and exit
+	-s                try to unpack openmpi.spec file from <tarball>
+
+	<tarball>         path to OpenMPI source tarball
+  "
+
+#
+# parse args
+#
+while getopts bn:omduf:t:sh flag; do
+    case "$flag" in
+      b) build_srpm="no"
+         build_single="yes"
+         ;;
+      n) rpmbuild_options="$rpmbuild_options --define '_name $OPTARG'"
+         ;;
+      o) rpmbuild_options="$rpmbuild_options --define 'install_in_opt 1'"
+	 configure_options="$configure_options --enable-mpirun-prefix-by-default --enable-orterun-prefix-by-default"
+         ;;
+      m) rpmbuild_options="$rpmbuild_options --define 'install_modulefile 1'"
+         ;;
+      d) rpmbuild_options="$rpmbuild_options --define 'build_debuginfo_rpm 1'"
+         ;;
+      u) configure_options="$configure_options --with-usnic --without-memory-manager --enable-mca-no-build=btl-openib,common-verbs,oob-ud --enable-mpi-ext=all"
+         ;;
+      f) configure_options="$configure_options --with-libfabric=$OPTARG \"LDFLAGS=-Wl,--build-id -Wl,-rpath -Wl,$OPTARG/lib64 -Wl,--enable-new-dtags\""
```

Comment:
In the Open MPI configury, we handle this complicated/sucky situation, but it takes a bunch of code.  😦 

* Test for the presence of `$dir/lib64`.  If that's there, use it.
* Test for the presence of `$dir/lib`.  If that's there, use it.
* If you didn't find either of them, error.

This is not bullet proof, either, because the above scheme clearly assumes a 64 bit build.  But what if the user also passed in relevant compiler/linker options to build in 32 bit?  It's a rat hole.  ☹️   So Open MPI *also* has a secondary, optional param for each package.  See https://github.com/open-mpi/ompi/blob/master/README#L874-L887.